pkg/operator/resource/resourceapply/core.go

```diff
@@ -337,23 +337,6 @@ func ApplyConfigMapImproved(ctx context.Context, client coreclientv1.ConfigMapsG
 
 // ApplySecret merges objectmeta, requires data
 func ApplySecretImproved(ctx context.Context, client coreclientv1.SecretsGetter, recorder events.Recorder, requiredInput *corev1.Secret, cache ResourceCache) (*corev1.Secret, bool, error) {
-	existing, err := client.Secrets(requiredInput.Namespace).Get(ctx, requiredInput.Name, metav1.GetOptions{})
```

Comment:
```go
existing, err := 
if err != nil && !apierrors.IsNotFound(err){
    return nil, false, err
}
if cache.SafeToSkipApply(requiredInput, existing) {
	return existing, false, nil
}

// do the required deep copy and string data stuff

if apierrors.IsNotFound(err) {
		requiredCopy := requiredInput.DeepCopy()
		actual, err := client.Secrets(requiredCopy.Namespace).
			Create(ctx, resourcemerge.WithCleanLabelsAndAnnotations(requiredCopy).(*corev1.Secret), metav1.CreateOptions{})
		reportCreateEvent(recorder, requiredCopy, err)
		cache.UpdateCachedResourceMetadata(requiredInput, actual)
		return actual, true, err
	}

// make sure all UpdateCachedResourceMetadata use requiredInput and not required
```

and you keep your "skip deep copy" behavior, right?

Reply:
I just tested your suggestion, but I had to make a small change to get the unit tests to pass (`requiredCopy := required.DeepCopy()` instead of `requiredCopy := requiredInput.DeepCopy()`